common/src/main/java/com/wynntils/features/user/tooltips/ItemCompareFeature.java

```diff
@@ -0,0 +1,165 @@
+import com.wynntils.mc.event.ScreenClosedEvent;
+import com.wynntils.mc.event.SlotRenderEvent;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.utils.McUtils;
+import com.wynntils.utils.objects.CommonColors;
+import com.wynntils.wc.custom.item.GearItemStack;
+import java.util.List;
+import java.util.Optional;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.client.gui.screens.inventory.AbstractContainerScreen;
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@FeatureInfo(stability = FeatureInfo.Stability.STABLE, category = "Item Tooltips")
+public class ItemCompareFeature extends UserFeature {
+    @RegisterKeyBind
+    private final KeyHolder toggleCompareModeKeybind =
+            new KeyHolder("Compare mode", GLFW.GLFW_KEY_SPACE, "Wynntils", true, () -> {});
+
+    @RegisterKeyBind
+    private final KeyHolder compareSelectKeybind =
+            new KeyHolder("Select for comparing", GLFW.GLFW_KEY_C, "Wynntils", true, () -> {});
+
+    private static GearItemStack COMPARED_ITEM = null;
+    private static boolean COMPARE_MODE_ON = false;
+
+    @SubscribeEvent
+    public void onRenderSlot(SlotRenderEvent.Pre event) {
+        Slot slot = event.getSlot();
+        if (slot.getItem() == COMPARED_ITEM) {
+            RenderUtils.drawArc(CommonColors.LIGHT_BLUE, slot.x, slot.y, 200, 1, 6, 8);
+        }
+    }
+
+    @SubscribeEvent
+    public void onItemTooltipRenderEvent(ItemTooltipRenderEvent.Post event) {
+        if (!(McUtils.mc().screen instanceof AbstractContainerScreen<?> abstractContainerScreen)) return;
+        if (!COMPARE_MODE_ON) return;
+
+        if (abstractContainerScreen.hoveredSlot != null) {
+            ItemStack hoveredItemStack = abstractContainerScreen.hoveredSlot.getItem();
+
+            if (event.getItemStack() != hoveredItemStack) {
+                return;
+            }
+
+            if (!(hoveredItemStack instanceof GearItemStack hoveredGearItemStack)) {
+                return;
+            }
+
+            // No compared item selected, try compare to equipped armor
+            if (COMPARED_ITEM == null) {
+                List<ItemStack> armorSlots = McUtils.player().getInventory().armor;
+
+                Optional<GearItemStack> matchingArmorItemStack = armorSlots.stream()
+                        .filter(itemStack -> itemStack instanceof GearItemStack gItemStack
+                                && gItemStack.getItemProfile().getItemInfo().getType()
+                                        == hoveredGearItemStack
+                                                .getItemProfile()
+                                                .getItemInfo()
+                                                .getType())
+                        .map(itemStack -> (GearItemStack) itemStack)
+                        .findFirst();
```

Comment:
Would there be a better approach for rings? Arrow keys to switch between them 🤔

Reply:
I think coding that might not be the most clean code for little benefit.